app/services/ClaimToAdjustService.scala

```diff
@@ -86,4 +89,67 @@ class ClaimToAdjustService @Inject()(val financialDetailsConnector: FinancialDet
     }).map(_.flatten)
   }
 
+  def getPoaTaxYearForEntryPoint(nino: Nino)(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[Throwable, Option[TaxYear]]] = {
+    checkCrystallisation(nino, getPoaAdjustableTaxYears).flatMap {
+      case None => Future.successful(Right(None))
+      case Some(taxYear: TaxYear) => financialDetailsConnector.getFinancialDetails(taxYear.endYear, nino.value).map {
+        case financialDetails: FinancialDetailsModel => Right(arePoAPaymentsPresent(financialDetails.documentDetails))
+        case error: FinancialDetailsErrorModel if error.code != NOT_FOUND => Left(new Exception("There was an error whilst fetching financial details data"))
+        case _ => Right(None)
+      }
+    }
+  }
+
+  private def arePoAPaymentsPresent(documentDetails: List[DocumentDetail]): Option[TaxYear] = {
+    documentDetails.filter(_.documentDescription.exists(description => description.equals("ITSA- POA 1") || description.equals("ITSA - POA ")))
+      .sortBy(_.taxYear).reverse.headOption.map(doc => makeTaxYearWithEndYear(doc.taxYear))
+  }
+
+  private def getPoaAdjustableTaxYears: List[TaxYear] = {
+    if (dateService.isAfterTaxReturnDeadlineButBeforeTaxYearEnd) {
+      List(
+        TaxYear.makeTaxYearWithEndYear(dateService.getCurrentTaxYearEnd)
+      )
+    } else {
+      List(
+        TaxYear.makeTaxYearWithEndYear(dateService.getCurrentTaxYearEnd).addYears(-1),
+        TaxYear.makeTaxYearWithEndYear(dateService.getCurrentTaxYearEnd)
+      ).sortBy(_.endYear)
+    }
+  }
+
+  private def checkCrystallisation(nino: Nino, taxYearList: List[TaxYear])(implicit hc: HeaderCarrier): Future[Option[TaxYear]] = {
+    taxYearList match {
+      case ::(head, next) => isTaxYearCrystallised(head, nino).flatMap {
+        case true => Future.successful(Some(head))
+        case false => checkCrystallisation(nino, next)
+      }
+      case Nil => Future.successful(None)
+    }
+  }
+
+  // Next 2 methods taken from Calculation List Service (with small changes), but I don't like the code duplication. Is there any solution to this?
+  private def isTaxYearCrystallised(taxYear: TaxYear, nino: Nino)(implicit hc: HeaderCarrier): Future[Boolean] = {
+
+    val currentTaxYearEnd = dateService.getCurrentTaxYearEnd
+    val futureTaxYear = taxYear.endYear >= currentTaxYearEnd
+    if (futureTaxYear) {
+      Future.successful(false)
+    } else {
+      getTYSCrystallisationResult(nino, taxYear.endYear)
+    }
+  }
+
+  private def getTYSCrystallisationResult(nino: Nino, taxYear: Int)(implicit hc: HeaderCarrier): Future[Boolean] = {
+    val taxYearRange = s"${(taxYear - 1).toString.substring(2)}-${taxYear.toString.substring(2)}"
```

Comment:
I guess this taxYear format can be provided by taxYear if we would change it to be type of TaxYear, but we need to check if we have exact format on the level of TaxYear type